app/scripts/controllers/service.js

```diff
@@ -60,6 +61,50 @@ angular.module('openshiftConsole')
           }
         );
 
+        watches.push(DataService.watch("services", context, function(services) {
+          $scope.services = services.by("metadata.name");
+        }));
+
+        watches.push(DataService.watch("pods", context, function(pods) {
+          var service = $scope.service;
+          if (!service) {
```

Comment:
dont request the service here. Instead call getPodsForService either way, and that function should just return if the service hasn't been loaded yet.  The serviceResolved method should also call getPodsForService, this way if the service loads after the pods the association will still happen.